SingularityUI/app/components/requestForm/RequestForm.jsx

```diff
@@ -0,0 +1,560 @@
+              <div className="alert alert-info alert-slim" role="alert">
+                <strong>Note:</strong> changes made below will only affect new tasks
+              </div> :
+              undefined}
+            <div className="form-group slavePlacement">
+              <label htmlFor="slavePlacement" className="control-label">
+                Slave Placement
+              </label>
+              <DropDown
+                id = 'slavePlacement'
+                prop = {{
+                  updateFn: event => {
+                    this.updateField('slavePlacement', event.target.value);
+                  },
+                  forceChooseValue: true,
+                  choices: [
+                    {
+                      value: "",
+                      user: "Default"
+                    },
+                    {
+                      value: "SEPARATE",
+                      user: "Separate"
+                    },
+                    {
+                      value: "OPTIMISTIC",
+                      user: "Optimistic"
+                    },
+                    {
+                      value: "GREEDY",
+                      user: "Greedy"
+                    }
+                  ],
+                  value: this.getValue('slavePlacement') ? this.getValue('slavePlacement') : ""
+                }}
+              />
+            </div>
+            {this.renderRequestTypeSpecificFormFields()}
+            <div id="button-row">
+              <span>
+                <button type="submit" className="btn btn-success btn-lg" disabled={this.cantSubmit() ? 'disabled' : undefined}>
+                  Save
+                </button>
+              </span>
+            </div>
+            {this.props.saveApiCall.error ?
+              <p className='alert alert-danger'>
+                There was a problem saving your request: {this.props.saveApiCall.error.message}
+              </p> :
+              undefined
+            }
+            {this.props.saveApiCall.data && this.props.saveApiCall.data.message ?
+              <p className='alert alert-danger'>
+                There was a problem saving your request: {this.props.saveApiCall.data.message}
+              </p> :
+              undefined
+            }
+          </form>
+        </div>
+      </div>
+    );
+  }
+
+};
+
+function mapStateToProps(state) {
+  return {
+    racks: state.api.racks.data,
+    request: state.api.request ? state.api.request.data : undefined,
+    form: state.form[FORM_ID],
+    saveApiCall: state.api.saveRequest
+  }
+}
+
+function mapDispatchToProps(dispatch) {
+  return {
+    update(formId, fieldId, newValue) {
+      dispatch(modifyField(formId, fieldId, newValue));
+    },
+    clearForm(formId) {
+      dispatch(clearForm(formId));
+    },
+    save(requestBody) {
+      dispatch(SaveAction.trigger(requestBody));
+    }
+  }
+}
+
+export default connect(mapStateToProps, mapDispatchToProps, undefined, {pure: false})(RequestForm);
```

Comment:
Gotcha, I'm okay with `pure: false` as long as we implement our own `shouldComponentUpdate` that does the right thing